src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -700,20 +700,20 @@ default StudentParticipation findByIdWithLegalSubmissionsElseThrow(long particip
      * Get all participations belonging to exam with submissions and their relevant results.
      *
      * @param examId the id of the exam
-     * @return list of participations belonging to course
+     * @return an unmodifiable list of participations belonging to course
      */
     default List<StudentParticipation> findByExamIdWithSubmissionRelevantResult(Long examId) {
         var participations = findByExamIdWithEagerLegalSubmissionsRatedResults(examId); // without test run participations
         // filter out the participations of test runs which can only be made by instructors
-        participations = participations.stream().filter(studentParticipation -> !studentParticipation.isTestRun()).collect(Collectors.toList());
+        participations = participations.stream().filter(studentParticipation -> !studentParticipation.isTestRun()).collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Since filterParticipationsWithRelevantResults() directly calls .steam() it should be fine to use a unmodifyable list here.
```suggestion
        participations = participations.stream().filter(studentParticipation -> !studentParticipation.isTestRun()).toList();
```

Reply:
You're right